provider/provider.go

```diff
@@ -14,9 +14,24 @@
 
 package provider
 
-import "context"
+import (
+	"errors"
+	"os"
 
-// Provider is an interface for securely loading secrets based on environment variables.
-type Provider interface {
-	LoadSecrets(ctx context.Context, paths []string) ([]string, error)
+	"github.com/bank-vaults/secret-init/model"
+	"github.com/bank-vaults/secret-init/provider/file"
+)
+
+func New(providerName string) (model.Provider, error) {
+	switch providerName {
+	case file.ProviderName:
+		provider, err := file.NewProvider(os.Getenv("SECRETS_FILE_PATH"))
+		if err != nil {
+			return nil, err
+		}
+		return provider, nil
+
+	default:
+		return nil, errors.New("invalid provider specified")
+	}
```

Comment:
IMO this should go to the main package for now. Generally lower level packages in the hierarchy should depend on higher level packages, not the other way around.